java8/src/main/java/io/cucumber/java8/Java8ParameterTypeDefinition.java

```diff
@@ -0,0 +1,26 @@
+package io.cucumber.java8;
+
+import io.cucumber.core.backend.ParameterTypeDefinition;
+import io.cucumber.cucumberexpressions.ParameterType;
+import java.util.Collections;
+import net.jodah.typetools.TypeResolver;
+
+class Java8ParameterTypeDefinition extends AbstractGlueDefinition implements ParameterTypeDefinition {
+
+    private final ParameterType parameterType;
+
+    @Override
+    public ParameterType<?> parameterType() {
+        return parameterType;
+    }
+
+    <T extends ParameterDefinitionBody> Java8ParameterTypeDefinition(String name, String regex, Class<T> bodyClass, T body) {
+        super(body, new Exception().getStackTrace()[3]);
+        Class returnType = TypeResolver.resolveRawArguments(ParameterDefinitionBody.class, bodyClass)[0];
```

Comment:
`returnType` here keeps resolving to Unknown, leading to:
```
io.cucumber.core.exception.CucumberException: Could not convert arguments for step [balloon coordinates {point}, defined by lambda] defined at 'TypeDefinitionsStepdefs.java:43'.
The details are in the stacktrace below.
	at io.cucumber.core.runner.PickleStepDefinitionMatch.couldNotConvertArguments(PickleStepDefinitionMatch.java:104)
	at io.cucumber.core.runner.PickleStepDefinitionMatch.runStep(PickleStepDefinitionMatch.java:48)
...
Caused by: io.cucumber.cucumberexpressions.CucumberExpressionException: ParameterType {point} failed to transform [3, 4] to class net.jodah.typetools.TypeResolver$Unknown
	at io.cucumber.cucumberexpressions.ParameterType.transform(ParameterType.java:237)
	at io.cucumber.cucumberexpressions.Argument.getValue(Argument.java:67)
	at io.cucumber.core.stepexpression.ExpressionArgument.getValue(ExpressionArgument.java:15)
	at io.cucumber.core.runner.PickleStepDefinitionMatch.runStep(PickleStepDefinitionMatch.java:43)
	... 49 more
Caused by: io.cucumber.core.backend.CucumberInvocationTargetException
	at io.cucumber.java8.Invoker.invoke(Invoker.java:24)
	at io.cucumber.java8.Java8ParameterTypeDefinition.execute(Java8ParameterTypeDefinition.java:24)
	at io.cucumber.cucumberexpressions.ParameterType.transform(ParameterType.java:233)
	... 52 more
```

Reply:
While the tests seem to work with `StringBuilder` and `StringBuffer`, I had no such luck when trying `Point`. Not sure if it works or is still broken with any non-string-y types.

Reply:
Pretty much confirmed now: it fails to parse anything but String arguments; It can return any type already though.

Reply:
Took another look at ParameterType both in cucumber-java and cucumber-expressions; it seems only handling String arguments might just be good enough, which simplifies things.